src/Utilities/cachingoptimizer.jl

```diff
@@ -148,16 +148,57 @@ function attach_optimizer(model::CachingOptimizer)
     # We do not need to copy names because name-related operations are handled by `m.model_cache`
     indexmap = MOI.copy_to(model.optimizer, model.model_cache, copy_names=false)
     model.state = ATTACHED_OPTIMIZER
-    # MOI does not define the type of index_map, so we have to copy it into a
-    # concrete container. Also load the reverse map.
-    model.model_to_optimizer_map = IndexMap()
-    model.optimizer_to_model_map = IndexMap()
-    for k in keys(indexmap)
-        model.model_to_optimizer_map[k] = indexmap[k]
-        model.optimizer_to_model_map[indexmap[k]] = k
+    # MOI does not define the type of index_map, so we might have to copy it
+    # into a concrete container. Also load the reverse map.
+    model.model_to_optimizer_map = convert(IndexMap, indexmap)#IndexMap()
+    model.optimizer_to_model_map = reverse_index_map(indexmap)#IndexMap()
+    nothing
+end
+
+function reverse_index_map(src::IndexMap)
+    dest = IndexMap()
+    sizehint!(dest.varmap, length(src.varmap))
+    # sizehint!(dest.conmap, length(src.conmap))
+    reverse_dict(dest.varmap, src.varmap)
+    reverse_dict(dest.conmap, src.conmap)
+    return dest
+end
+function reverse_dict(dest::AbstractDict, src::AbstractDict)
```

Comment:
These methods should begin with underscores. And this one could do with a docstring that it assumes the values are unique.